### ISSUE TITLE
Joined table inheritance with AttributeOverride (rename) of the identifier column fails on insert

### ISSUE THREAD

**Issue by cziegenberg** [doctrine/orm#8229]
In the following example, the automatically created insert into the resource table fails when creating a new user, because the id field got another column name in the user table, using the AttributeOverride annotation:

// src/Entity/Resource.php
```
/**
 * @ORM\Entity()
 * @ORM\Table(name="app_resource")
 * @ORM\InheritanceType("JOINED")
 * @ORM\DiscriminatorColumn(name="resource_type", type="string", length=191)
 * @ORM\DiscriminatorMap({"resource"="Resource", "user"="User"})
 */
abstract class Resource
{
    /**
     * @ORM\Id()
     * @ORM\Column(name="resource_id", type="guid")
     */
    protected string $id;
}
```

// src/Entity/User.php
```
/**
 * @ORM\Entity()
 * @ORM\Table(name="app_user")
 * @ORM\AttributeOverrides({@ORM\AttributeOverride(name="id", column=@ORM\Column(name="user_id", type="guid"))})
 */
class User extends Resource
{
    //...
}
```

Error (meaning only one parameter instead of two given for the prepared statement, because the id value is not set):
```
  [Doctrine\DBAL\Driver\PDOException (08P01)]                                                                                      
  SQLSTATE[08P01]: <<Unknown error>>: 7 FEHLER:  Binden-Nachricht enthält 1 Parameter, aber vorbereitete Anweisung »« erfordert 2 
```

Also reading the entity, updating it and deleting it fails. I have a working fix for it, tested with the 2.8.x and the 2.7.x branch.

**Comment by cziegenberg** [comment]
That's not enough. There are other parts effected by this (imho) bug, i.e. when creating joins to the root table... I will have a look into it, but I'm totally new to Doctrine, so it could take some time. Is it generally allowed to create a pull request for a 2.x branch for this type of fix or would this be treated as a new feature?

**Comment by cziegenberg** [comment]
Also fixed in master. See pull requests.

**Comment by greg0ire** [comment]
Reopening since it isn't fixed until your PR is merged.

**Comment by rogregoire** [comment]
@ostrolucky @greg0ire : 
Is it possible that the fix, and the associated PR, created a bug/regression in BasicEntityPersister::getSelectConditionStatementSQL for entities that use ResolveTargetEntities and have class::inheritanceType value INHERITANCE_TYPE_NONE which goes in the if(isset($this->class->fieldMappings[$field]['inherited']) but shouldn't ?

**Comment by greg0ire** [comment]
Anything is possible. If you want to be sure you can try reverse-applying [the patch of the MR](https://patch-diff.githubusercontent.com/raw/doctrine/orm/pull/8234.patch) (`patch` has a `--reverse` option)

**Comment by rogregoire** [comment]
I already xdebug many hours to understand my bug to be originated to this. 
What I don't know if it's a bug or a feature. I don't have a deep enough knowledge of doctrine 2.x (stayed 10 years in 1.2...)

So my previous comment shoud be: does "INHERITANCE_TYPE_NONE" class should go in that if ? (because it does)

My bug is solved by replacing 
`            if (isset($this->class->fieldMappings[$field]['inherited'])) {`
by 
`            if (isset($this->class->fieldMappings[$field]['inherited']) && $this->class->inheritanceType !== ClassMetadata::INHERITANCE_TYPE_NONE) {
`
But i've no idea if i've should have done that and if it could break anything else...

The context is:
I've an Entity `Uai` that replace a class `VUai` through ResolveTargetEntities.
A Select SQL query is badly generated because it in the where part, it generates a new alias, because that if use the wrong classname:

`An exception occurred while executing 'SELECT t1.uairne AS uairne_2, t1.uainum AS uainum_3, [...] FROM DB2_MYSQL_V_UAI t1 WHERE t0.uairne = ?' with params ["0070381B"]:`

obviously, the t0 and t1 should be the same alias. but that IF sets the `abstract` replaced classname, so `getSQLTableAlias` generates a new one

I guess that the PR did not take in account the `INHERITANCE_TYPE_NONE`, but once again, it's a very wild guess, I'm really a noob in doctrine 2.

**Comment by cziegenberg** [comment]
@rogregoire Doctrine ORM is also very new to me. I fixed this problem to be able to use Doctrine for my first project. :) So it took a lot of time to work into it, but Doctrine ORM is complex (especially all these arrays, which aren't completly documented) and I would never say that I covered all use cases - I tried to, but in the end I have to trust the tests.

So yes, perhaps this use case is missing. I cannot say if your change breaks anything, if so, this should be reported by a failing test. :)
